src/letsbuilda/pypi/models/models_json.py

```diff
@@ -114,10 +114,43 @@ class Info:
     version: str
     yanked: bool
     yanked_reason: str | None
+    dynamic: (
+        list[
+            Literal[
+                "Platform",
+                "Supported-Platform",
+                "Summary",
+                "Description",
+                "Description-Content-Type",
+                "Keywords",
+                "Home-page",
+                "Download-URL",
+                "Author",
+                "Author-email",
+                "Maintainer",
+                "Maintainer-email",
+                "License",
+                "Classifier",
+                "Requires-Dist",
+                "Requires-Python",
+                "Requires-External",
+                "Project-URL",
+                "Provides-Extra",
+                "Provides-Dist",
+                "Obsoletes-Dist",
+            ]
+        ]
+        | None
+    )
+    provides_extra: list[str] | None
 
     @classmethod
     def from_dict(cls: type[Self], data: dict) -> Self:  # type: ignore[type-arg]
         """Build an instance from a dictionary."""
+        if "dynamic" not in data:
+            data["dynamic"] = None
+        if "provides_extra" not in data:
+            data["provides_extra"] = None
```

Comment:
Despite the code saying that [the attributes are generated at request time](https://github.com/pypi/warehouse/blob/main/warehouse/legacy/api/json.py#L167), they don't seem to actually always be present.

For example `dynamic` _is_ present in https://pypi.org/pypi/dspygen/json but is _not_ present in https://pypi.org/pypi/letsbuilda-pypi/json.